Update fips-images.md for kernel independent fips (#1912)
chainguard-dev/edu: content/chainguard/chainguard-images/working-with-images/fips-images.md

---

> title: "Chainguard's Kernel Independent FIPS-Enabled Images"

Reviewer: Can we use the actual product title here? "Chainguard FIPS Images" is the term used on the website, order sheets, legal documents, contracts, all other marketing?

linktitle: "FIPS Images"
aliases: 
- /chainguard/chainguard-images/fips-images
- /chainguard/chainguard-images/images-features/fips-images
type: "article"
description: "A conceptual overview of Chainguard's Kernel Independent FIPS-Enabled Images."

> date: 2024-02-08T15:56:52-07:00

Reviewer: "Chainguard FIPS Images" like in the title.

Chainguard FIPS Images remove this friction with a novel design that replaces a kernel entropy source with a userspace one. This implementation enables developers to deploy FIPS workloads using any of the latest kernels, hardware, and instance types. Chainguard FIPS Images thus unlock the ability to run FIPS workloads on developer machines, existing CI/CD deployments, and even on readily available non-FIPS managed cloud offerings. All this can be done using the latest userspace runtimes like NodeJS, Python, Go, PHP, .NET, and C/C++, among others. Under Chainguard’s novel design, the container image for a given FIPS application can be entirely self-contained, minimal, and distroless.

## Available FIPS Images

> 

Reviewer: this section reffers to extensive additional technical details. This is not going to be a list of 300+ images we have =)

So a better sub-tittle is needed here "Developer Guidance", "Deployment Manual", "Developer Guides" maybe?

Additional guidance is available for specific images, like these:

- [go-fips](https://images.chainguard.dev/directory/image/go-fips/overview)
- [node-fips](https://images.chainguard.dev/directory/image/node-fips/overview)
- [jdk-fips](https://images.chainguard.dev/directory/image/jdk-fips/overview)

You can a full list Chainguard's FIPS Images at: [https://images.chainguard.dev/?category=fips](https://images.chainguard.dev/?category=fips).

> 

Reviewer: "Chainguard FIPS Images"

toc: true
---

## What is FIPS? 

> One of the primary requirements of federal compliance frameworks — including [FedRAMP](https://www.fedramp.gov/program-basics/) — is to use FIPS-validated cryptography. To help customers meet these requirements, Chainguard offers FIPS-enabled versions of many images. This article provides a high-level overview of what FIPS is, what to expect from Chainguard’s FIPS Images, and how Chainguard's Kernel-Independent FIPS images stand out from alternatives.

Reviewer 2: I like that.

> One of the primary requirements of federal compliance frameworks — including [FedRAMP](https://www.fedramp.gov/program-basics/) — is to use FIPS-validated cryptography. To help customers meet these requirements, Chainguard offers FIPS-enabled versions of many images. This article provides a high-level overview of what FIPS is, what to expect from Chainguard’s FIPS Images, and how Chainguard's Kernel-Independent FIPS images stand out from alternatives.

Reviewer 2: ```suggestion
One of the primary requirements of federal compliance frameworks — including [FedRAMP](https://www.fedramp.gov/program-basics/) — is to use FIPS-validated cryptography. To help customers meet these requirements, Chainguard offers FIPS-enabled versions of many images. This article provides a high-level overview of what FIPS is, what to expect from Chainguard FIPS Images, featuring a kernel-independent design, and how Chainguard FIPS images stand out from alternatives.
```

## What To Expect from Chainguard FIPS Images

‍Chainguard warranties the following with respect to Chainguard container images:

Chainguard FIPS Images available to be delivered in compliance with FIPS specifications are listed [here](https://images.chainguard.dev/?category=fips)  (each a "Chainguard FIPS Image"). Images will be made available in compliance with FIPS specifications provided a customer’s applicable order form designates the purchase of Chainguard FIPS images.

The Chainguard FIPS images contain FIPS-validated software cryptographic modules. Entropy must be provided as specified in its cryptographic policy. The cryptographic module may provide non-approved algorithms, which will result in operating in FIPS non-approved mode. The cryptographic FIPS modules currently provided are:

> - OpenSSL FIPS 3.0 Provider Module (CVMP [#4856] (https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4856))

Reviewer 3: BTW there's a rogue space after the ] which breaks the link

Reviewer 2: Thank you!

Reviewer 3: Another thing; I think it should be "Chainguard OpenSSL FIPS 3.0 Provider Module", just to make it clear that it's our own cert (unlike the bouncy castle ones). This also matches the text in the link.

> - OpenSSL FIPS 3.0 Provider Module (CVMP [#4856] (https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4856))

Reviewer 2: ```suggestion
- OpenSSL FIPS 3.0 Provider Module (CVMP [#4856] (https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4856))
```

- Bouncy Castle FIPS Java API (CMVP [#4743](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4743), CMVP [#4616](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4616))
- Chainguard CPU Time Jitter RNG Entropy Source (ESV Entropy Certificate [#E191](https://csrc.nist.gov/projects/cryptographic-module-validation-program/entropy-validations/certificate/191))

These modules may be updated occasionally; for further information, contact <fips-contact@chainguard.dev>.

## Chainguard's Kernel-Independent FIPS Images

> Cryptographic protection relies on the secure implementation of a trusted algorithm and a random bit generator that cannot be reasonably predicted at any greater accuracy than random chance. Traditionally, to achieve this compliance requirement, developers were required to provision dedicated hardware and VMs with the host kernel configured in FIPS mode. This is because container historically accessed the entropy source provided by a certified kernel. In cloud native or shared environments, this requirement significantly increased operational complexity by forcing a dependence on a limited set of FIPS-enabled kernels. 

Reviewer 2: ```suggestion
Cryptographic protection relies on the secure implementation of a trusted algorithm and a random bit generator that cannot be reasonably predicted at any greater accuracy than random chance. Traditionally, to achieve this compliance requirement, developers were required to provision dedicated hardware and VMs with the host kernel configured in FIPS mode. This is because containers historically accessed the entropy source provided by a certified kernel. In cloud native or shared environments, this requirement significantly increased operational complexity by forcing a dependence on a limited set of FIPS-enabled kernels. 
```

Note: There are some workloads that require a kernel SP 800-90B entropy source or a kernel FIPS module. These include but are not limited to Chainguard FIPS images shipping Java, k8s CNI plugins, LUKS2 full-disk encryption, and StrongSwan VPN. These use cases will continue to require a kernel in FIPS mode.

Read our full blog about Chainguard's Kernel-Independent FIPS Images at: [https://www.chainguard.dev/unchained/kernel-independent-fips-images].

## Developer Guidance for Available FIPS Images

Additional guidance is available for specific images, like these:

- [go-fips](https://images.chainguard.dev/directory/image/go-fips/overview)
- [node-fips](https://images.chainguard.dev/directory/image/node-fips/overview)
- [jdk-fips](https://images.chainguard.dev/directory/image/jdk-fips/overview)

> You can a full list Chainguard's FIPS Images at: [https://images.chainguard.dev/?category=fips](https://images.chainguard.dev/?category=fips).

Reviewer 2: ```suggestion
You can find a full list Chainguard FIPS Images at: [https://images.chainguard.dev/?category=fips](https://images.chainguard.dev/?category=fips).
```
